pyproject.toml

```
@@ -1,3 +1,77 @@
 [build-system]
-requires = ["setuptools>=40.8.0", "wheel"]
+requires = ["setuptools", "setuptools-scm", "wheel"]
 build-backend = "setuptools.build_meta"
+
+
+[project]
+name = "zarr"
+description = "An implementation of chunked, compressed, N-dimensional arrays for Python"
+readme = { file = "README.md", content-type = "text/markdown" }
+maintainers = [ {name = "Alistair Miles", email = "alimanfoo@googlemail.com"} ]
+requires-python = ">=3.8"
+dependencies = [
+    'asciitree',
+    'numpy>=1.7',
+    'fasteners',
+    'numcodecs>=0.10.0',
+]
+dynamic = [
+  "version",
+]
+classifiers = [
+    'Development Status :: 6 - Mature',
+    'Intended Audience :: Developers',
+    'Intended Audience :: Information Technology',
+    'Intended Audience :: Science/Research',
+    'License :: OSI Approved :: MIT License',
+    'Programming Language :: Python',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+    'Operating System :: Unix',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+]
+
+[project.optional-dependencies]
+jupyter = [
+    'notebook',
+    'ipytree',
+]
```

Heads up. This just got [updated]( https://github.com/zarr-developers/zarr-python/blob/f9581cb6db4c2950e200add3785bac2685aac391/setup.py#L31-L35 )

```suggestion
[project.optional-dependencies]
jupyter = [
    'notebook',
    'ipytree>=0.2.2',
    'ipywidgets>=8.0.0',
]
```